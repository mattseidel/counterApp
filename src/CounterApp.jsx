import React, { useState } from "react";
import PropTypes from "prop-types";
import "./index.css";
const CounterApp = ({ value = 10 }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = (e) => {
    e.preventDefault();
    setCounter(counter + 1);
  };
  const handleReset = (e) => {
    e.preventDefault();
    setCounter(value);
  };
  const handleMinus = (e) => {
    e.preventDefault();
    setCounter(counter - 1);
  };
  CounterApp.protoType = { value: PropTypes.number };
  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleReset}>reset</button>
      <button onClick={handleMinus}>-1</button>
    </>
  );
};

export default CounterApp;
