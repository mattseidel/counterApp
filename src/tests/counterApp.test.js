import React from 'react'

import { shallow } from 'enzyme'
import '@testing-library/jest-dom'
import CounterApp from '../CounterApp'
describe('test on PrimeraApp', () => {

    let wrapper = shallow(<CounterApp />)
    beforeEach(() => {
        wrapper = shallow(<CounterApp />)
    })

    test('should component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should show value correctly', () => {

        const value = "100"
        const wrapper = shallow(
            <CounterApp
                value={value}
            />)
        const textoParrafo = wrapper.find('h2').text()
        expect(textoParrafo).toBe(value)

    })

    test('should increment with +1 button', () => {
        wrapper.find("button").at(0).simulate('click', { preventDefault: () => { } });
        const textoParrafo = wrapper.find('h2').text()
        console.log(textoParrafo);
        expect(textoParrafo).toBe("11")

    })

    test('should increment with -1 button', () => {
        wrapper.find("button").at(2).simulate('click', { preventDefault: () => { } });
        const textoParrafo = wrapper.find('h2').text()
        console.log(textoParrafo);
        expect(textoParrafo).toBe("9")

    })

    test('should reset default value', () => {
        const wrapper = shallow(
            <CounterApp
                value={105}
            />)
        wrapper.find("button").at(0).simulate('click', { preventDefault: () => { } });
        wrapper.find("button").at(0).simulate('click', { preventDefault: () => { } });
        wrapper.find("button").at(1).simulate('click', { preventDefault: () => { } });
        const textoParrafo = wrapper.find('h2').text()

        expect(textoParrafo).toBe('105')

    })


})