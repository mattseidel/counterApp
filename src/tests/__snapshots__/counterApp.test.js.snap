// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on PrimeraApp should component correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
